libs/contrib/Decidable/Decidable/Extra.idr

```diff
@@ -47,3 +51,23 @@ doubleNegationExists {ts} {r} @{dec} nnxs =
       witnessing   : uncurry              r  witness
       witnessing   = doubleNegationElimination @{dec} witness witnessingnn
   in introduceWitness witness witnessing
+
+
+||| Convert a decision about a decidable property into one about its negation.
+public export
+negateDec : (1 dec : Dec a) -> Dec (Not a)
+negateDec (Yes pf) = No ($ pf)
+negateDec (No npf) = Yes npf
+
+
+||| If a relation is decidable, then so is its complement
+public export
+[DecidableComplement] {ts : Vect n Type} -> {r : Rel ts} -> (posDec : Decidable n ts r) =>
+  Decidable n ts (complement {ts} r) where
```

Comment:
Why not go via `curry/uncurry` and `notToComplement` & `complementToNot` lemmas?